velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1495,7 +1500,40 @@ struct ToISO8601Function {
   FOLLY_ALWAYS_INLINE void call(
       out_type<Varchar>& result,
       const arg_type<Date>& date) {
-    result = DateType::toIso8601(date);
+    result = DateType::toIso8601(date);    
+  }
+};
+template <typename T>
+struct TimestampWithTimezoneAtTimezoneFunction
+    : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  std::string sessionTzName = "";
+  const date::time_zone* sessionTimezone = nullptr;
+  int64_t targetTimezoneID = -1;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<TimestampWithTimezone>* /*tsWithTz*/,
+      const arg_type<Varchar>* timeZone) {
+    // below statement not working in init()
```

Comment:
See https://facebookincubator.github.io/velox/develop/scalar-functions.html section Access to Session Properties and Constant Inputs.
It likely depends on how the function is called to determine whether or not the timezone is a constant. When operating on a vector of values this might become a constant e.g. imagine selecting a column 
```
select col1 AT TIME ZONE 'America/Los_Angeles'
```
where col1 is a timestamp with time zone column. I'd imagine in that case the time zone is a constant where initialize could apply.
So I think you can add it here but also need to handle it in the call in case it was not set during initialize.

Reply:
Thanks for pointing this out Christian, makes sense. I noticed the example in the docs handling the possibility of a constant in initialize - modified my code here accordingly.